jmespath-jackson-jr/src/main/java/io/burt/jmespath/jacksonjr/JacksonJrRuntime.java

```diff
@@ -0,0 +1,225 @@
+package io.burt.jmespath.jacksonjr;
+
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.jr.ob.JSON;
+import com.fasterxml.jackson.jr.stree.JacksonJrsTreeCodec;
+import com.fasterxml.jackson.jr.stree.JrsArray;
+import com.fasterxml.jackson.jr.stree.JrsBoolean;
+import com.fasterxml.jackson.jr.stree.JrsNull;
+import com.fasterxml.jackson.jr.stree.JrsNumber;
+import com.fasterxml.jackson.jr.stree.JrsObject;
+import com.fasterxml.jackson.jr.stree.JrsString;
+import com.fasterxml.jackson.jr.stree.JrsValue;
+import io.burt.jmespath.BaseRuntime;
+import io.burt.jmespath.JmesPathType;
+import io.burt.jmespath.RuntimeConfiguration;
+import java.io.IOException;
+import java.util.AbstractList;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class JacksonJrRuntime extends BaseRuntime<TreeNode> {
```

Comment:
Most of the comments below were written before I came to the end and saw that the objects handled by this runtime adapter all seem to be subclasses of `JrsValue`. Most are still relevant, but some of the suggestions would probably be different depending on the answer to the following question:

Why is the base type `TreeNode` and not `JrsValue`? The latter seems to be much easier to code against, with less casting and less convoluted traversals through `JsonToken`. Is there any way that any object handled by this runtime adapter is _not_ an instance of `JrsValue`?

Reply:
Good catch! all implementations of `TreeNode` in jackson-jr are `JrsValue` or children.